src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientProxy.java

```diff
@@ -69,6 +78,91 @@ private BlobContainerClient get(TargetStorageAccount targetStorageAccount, Strin
         }
     }
 
+    public void streamContentToDestination(
+        BlobClient sourceBlob,
+        String destinationContainer,
+        TargetStorageAccount targetStorageAccount
+    ) {
+        try {
+            streamContent(sourceBlob.getBlockBlobClient(), destinationContainer, targetStorageAccount);
+        } catch (HttpResponseException ex) {
+            logger.info(
+                "Uploading failed for blob {} to Container: {},  error code: {}",
+                sourceBlob.getBlobName(),
+                destinationContainer,
+                ex.getResponse() == null ? ex.getMessage() : ex.getResponse().getStatusCode()
+            );
+            if ((targetStorageAccount == TargetStorageAccount.CFT
+                || targetStorageAccount == TargetStorageAccount.PCQ)
+                && HttpStatus.valueOf(ex.getResponse().getStatusCode()).is4xxClientError()) {
+                sasTokenCache.removeFromCache(destinationContainer);
+            }
+            throw ex;
+        }
+    }
+
+    private void streamContent(
+        BlockBlobClient sourceBlob,
+        String destinationContainer,
+        TargetStorageAccount targetStorageAccount
+    ) {
+        var blobName = sourceBlob.getBlobName();
+        logger.info("Start uploading from blob {} to Container: {}", sourceBlob.getBlobUrl(), destinationContainer);
+        long startTime = System.nanoTime();
+        try (var zipStream = new ZipInputStream(sourceBlob.openInputStream());) {
+            ZipEntry entry;
+
+            while ((entry = zipStream.getNextEntry()) != null) {
+                if (Objects.equals(entry.getName(), ENVELOPE)) {
+                    final BlockBlobClient blockBlobClient =
+                        get(targetStorageAccount, destinationContainer)
+                            .getBlobClient(blobName)
+                            .getBlockBlobClient();
+
+                    ParallelTransferOptions parallelTransferOptions =
```

Comment:
even we try to outstream api does not make the http request until it reaches its buffer limits so we need to reduce the limits otherwise it gets memory out of error.